common/app/routes/challenges/redux/fetch-challenges-saga.js

```diff
@@ -25,7 +25,7 @@ export default function fetchChallengesSaga(action$, getState, { services }) {
     ))
     .flatMap(({ type, payload: { dashedName, block } = {} }) => {
       const state = getState();
-      const lang = state.app.languageTag;
+      const lang = state.app.lang;
```

Comment:
This key is updated by the url change, not directly by the user. We always want the URL lang to take precedence over the user selected lang. 

On the settings page, when the user selects a lang, a redirect to the new selected lang happens which will indirectly to updating this var. After the redirect, then we refetch. This incorrect key is why it was initially broken.